view/src/storage.rs

```
@@ -117,9 +122,49 @@ impl Storage {
         Ok(Storage {
             pool,
             uncommitted_height: Arc::new(Mutex::new(None)),
+            scanned_notes_tx: broadcast::channel(10).0,
         })
     }
 
+    pub fn await_change(
+        &self,
+        note_commitment: tct::Commitment,
+    ) -> impl Future<Output = anyhow::Result<NoteRecord>> + '_ {
+        let mut rx = self.scanned_notes_tx.clone().subscribe();
+
+        async move {
+            match self.note_record_by_commitment(note_commitment).await? {
+                Some(record) => Ok(record),
+
+                None => loop {
+                    let record = rx.recv().await.unwrap();
```

Should we bubble up the error here?